### ISSUE TITLE
Race condition bug on "my-fueled-app" example

### ISSUE THREAD

**Issue by paulorobertorosa** [FuelLabs/fuels-ts#498]
### What version of fuels-ts are you using?

Latest (v0.13.0)

### Steps to Reproduce

1. Run the "my-fueled-app" sample application according to [Quickstart ](https://fuellabs.github.io/fuels-ts/QUICKSTART) guidelines walkthrough.
2. Click into the "Increment" button several times, the more fast you can. (You can also use a script to click on it very fast to make easier to reproduce)
3. Look into the console of the fuel-core API running, and you can see errors mentioning: "Transaction id was already used".

### Expected Behavior

Expected behavior is to use the cleanup function to prevent newer promises from being fired before older ones gets fulfilled.

Or it can also be a loading state, or even to keep track about last used transaction id, but i feel that's too much for a simple task.

### Actual Behavior

Basically the errors are popping up on the console of fuel-core API running. And this is because of promise race condition bug which occurs because the old promise was fulfilled before the newer promise. The application is not handling that edge case because there is no cleanup being used on hooks or there is no loading state or something like this.

**Comment by paulorobertorosa** [comment]
incorrect github account
